content-security-policy/script-src/script-src-strict_dynamic_double_policy_different_nonce.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>A separate policy with more nonces works correctly with `strict-dynamic` in the script-src directive.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+
+    <!-- CSP served:
+    1) Content-Security-Policy: script-src 'strict-dynamic' 'nonce-dummy'
+    2) Content-Security-Policy: script-src 'nonce-dummy' 'nonce-dummy2'
+    -->
+</head>
+
+<body>
+    <h1>A separate policy with more nonces works correctly with `strict-dynamic` in the script-src directive.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            var handler = function(e) {
+                assert_equals(document.getElementById('unNonced-appendChild').getAttribute('executed'), null);
+                assert_equals(e.violatedDirective.split(' ')[0], 'script-src');
+                document.removeEventListener('securitypolicyviolation', handler);
+                t.done();
+            };
+            document.addEventListener('securitypolicyviolation', handler);
+
+            var e = document.createElement('script');
+            e.id = 'unNonced-appendChild';
+            e.src = 'simpleSourcedScript.js';
+            e.onload = t.unreached_func('OnLoad should not be triggered.');
+            document.body.appendChild(e);
+        }, 'Unnonced script injected via `appendChild` is not allowed with `strict-dynamic` + a nonce-only double policy.');
+    </script>
+
+    <!-- This test must be synchronous because we have to wait until the old event handler is removed. -->
+    <script nonce='dummy'>
+        test(function(t) {
+            document.addEventListener('securitypolicyviolation', t.unreached_func('No CSP violation report has fired.'));
+
+            var e = document.createElement('script');
+            e.setAttribute('nonce', 'dummy2');
+            e.id = 'nonced-appendChild';
+            e.src = 'simpleSourcedScript.js';
+            e.onload = t.step_func_done(function() {
+                assert_equals(document.getElementById('nonced-appendChild').getAttribute('executed'), '1');
+            });
+            e.onerror = t.unreached_func('Error should not be triggered.');
+            document.body.appendChild(e);
```

Comment:
I am not sure I understand, I already have:

```
document.addEventListener('securitypolicyviolation', t.unreached_func('No CSP violation report has fired.'));
```

in the second test, and it is the only listener at that point. I want to make sure that does not trigger, and that's all. How is your solution different?